Execs for srcML Parsed Source Code #314

daomcgill opened the pull request:
Blocked by #320

daomcgill committed:
i #313 Added Notebook for Syntax Extraction

- Documented current syntax extraction functions
- Overview on syntax extraction and XPath
- Placeholder for syntax.yml config file

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #313 Added File-Level and Class-Level Doc Functions

- Added new functions
- New configuration file
- Updated documentation

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #313 Update Config File

- Remove unused settings
- Change ../ to ../../
- Update notebook to reflect changes

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #313 Updates to Syntax Extractor

- Added parameter for excluding licenses in class and file-level comment extraction
- Implemented function extraction for function names with optional parameters
- Implemented variable extraction with optional types
- Added examples for removing empty comments and/or comment delimiters

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #313 Add Function for Import Query

- Added function for imports
- Reformatted new query functions
- Added Notebook Example for Joined Queries

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #313 Add Rd File

daomcgill committed:
i #313 Fix for Query Functions

- Fix for issue with namespaces in certain queries
- TO DO: Package function currently missing filepath

Signed-off-by: Dao McGill

daomcgill committed:
i #313 Fix for Package Query

- Now displays filenames correctly

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #313 Fix Roxygen version

daomcgill committed:
i #313 Actions Fix Attempt

daomcgill committed:
Revert "i #313 Actions Fix Attempt"

This reverts commit 22d85f9bc84231374bc1a8418ce5bc783e967122.

daomcgill committed:
i #313 Actions Fix Attempt

daomcgill committed:
i #313 Actions Fix Attempt

daomcgill committed:
i #313 Notebook Revision

- TO DO: Cheatsheet for this work thread

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #313 Minor Fix: Change srcml_path to use tools.yml

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #314 Added exec/annotate.R

- WIP

Signed off by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #314 Added exec/annotate.R and exec/query.R

- annotate.R generates the XML file
- query.R calls the query functions, depending on options

Note: may have to revisit output format after getting into the fasttext notebook

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #314 Added class names to query.R

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

carlosparadis commented on exec/query.R:
```diff
@@ -23,6 +23,7 @@ OPTIONS:
   --packages       Run the packages query.
```
I would recommend being more descriptive here. What exactly is the user getting? So, id say something like:

"Returns a 3 column table (filepath, classpath, file_content), where every row is defined by a class docstring" or 

"Returns a 3 column table (filepath, classpath, file_content), where every row is defined by a class name"

etc. Remember the user will have no idea what "imports query" means.

carlosparadis commented on exec/query.R:
the name of the file should be updated to src_content_parser.R since it is basically a src code parser to extract content as tables.

carlosparadis commented on exec/query.R:
```diff
@@ -10,7 +10,7 @@ require(XML, quietly = TRUE)
+  query.R query <tools.yml> <project_conf.yml> <srcml_filepath> [--namespace | --file-docs | --class-docs | --variables | --packages | --functions | --imports | --class-names]
 
 DESCRIPTION:
   Runs a specified query on the annotated XML file and saves the output to the appropriate path.
```
I would be more verbose here and focus more on input and output and third party tool dependencies. 

Something like "Uses srcML to parse a src code folder, as specified on the project configuration file, to parse selected content as (file,class,variable) tables. "

carlosparadis reviewed (changes requested):
Some more interface documentation notes.

daomcgill committed:
i #314 Updates to Exec Scripts

- Renamed query.R to src_content_parser.R
- Edited description
- Added descriptions for options
- Changed output path slightly
- Added a temp config file for easy testing for fasttext issue
NOTE: current output_path is a temporary solution that is useful for me right now. This will be fixed pre-merge.

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

anthonyjlau committed:
i #230 create config file interface

This commits perform a major refactoring of how Kaiaulu interface with config files, and the suggested folder organization to store rawdata and analysis. 

The configuration files are generalized to account for anomaly cases when performing project analysis. For instance, long-lived projects may contain multiple repositories, issue trackers, mailing list, etc. The new template of the configuration file allows to account for this information. 

Moreover, changes to the config template cascaded in changes to all notebooks, as the access to the config was hardcoded to the file organization. A new set of get_ functions should make this the last commit that change in template cascades into notebooks. All actively maintained notebooks  (not prefixed by underline under vignettes/) have been updated to use the get functions. Future changes, therefore, will only affect the get() functions in R/config.R.

The folder organization of the filepaths has also been modified. Previously, filepaths assumed as default in the versioned config files suggested organizing code as rawdata/git_repo/projectX ; rawdata/jira/projectY. This organization was not practical for sharing data manually, as the user would need to zip several folders individually. The new organization is now rawdata/projectX/git_repo ; rawdata/projectX/jira. This means users only need to zip projectX and that will contain all the data wanted to be shared.

A minor typo on graph.R was also fixed for merge function calls from `sorted=` to `sort=`.

daomcgill committed:
Merge branch 'master' of https://github.com/sailuh/kaiaulu

daomcgill committed:
i #314 Exec Scripts Updates

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #314 Update exec/src_content_parser to work with code_embedding repo

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #314 Minor Config Changes

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

carlosparadis commented:
@daomcgill Heads-up: I am reviewing this one now, so don't commit to this PR to avoid merge conflict.

carlosparadis committed:
Merge branch 'master' into 314-externalize-parsed-source-code

carlosparadis commented:
Oops, it is #320 that I need to review first, these are the execs. Not locked anymore until I come back around this.

carlosparadis committed:
Merge branch 'master' into 313-expand-syntax-extractor

carlosparadis committed:
Undo some local changes

Signed-off-by: Carlos Paradis <carlosviansi@gmail.com>

carlosparadis commented:
@daomcgill hi Dao, I am unfortunately out of time to fix these. Could you add the get here? 

@crepesAlot @nicoelee123 @RavenMarQ please make sure as you code review each other, and your own code that you are consistently using the get, and that your *executable* scripts are too. Please take a look at the functions I already reviewed, especially the title. I removed the `@description` tag since Kaiaulu does not use it. 

It would be very helpful if you could sync to compare notes to apply the changes already merged consistently, since I had to re-apply a couple a few times (which unfortunately ate into the time to merge more PRs).

carlosparadis commented on .github/workflows/R-CMD-check.yml:
```diff
@@ -15,7 +15,7 @@ jobs:
     strategy:
       matrix:
-        r-version: ['4.4']
+        r-version: ['4.1']
```
Same issue again with workflow getting modified. @daomcgill @RavenMarQ @crepesAlot @beydlern make sure you account for this

carlosparadis commented on DESCRIPTION:
```diff
@@ -45,7 +45,7 @@ Imports:
     curl (>= 4.3),
     gh (>= 1.2.0),
-    XML (>= 3.99-0.7),
+    XML (>= 3.99),
```
again

carlosparadis commented on exec/annotate.R:
```diff
@@ -0,0 +1,61 @@
+  conf_path <- arguments[["<maven.yml>"]]
+  srcml_filepath <- arguments[["<srcml_filepath>"]]
+
+  tool <- yaml::read_yaml(tools_path)
```
hardcoded, need get

carlosparadis commented on exec/src_content_parser.R:
```diff
@@ -0,0 +1,104 @@
+output_dir <- path.expand(arguments[["<output_dir>"]])
+
+# Read configuration
+tool <- yaml::read_yaml(tools_path)
```
get again

carlosparadis commented on vignettes/text_gof_showcase.Rmd:
````diff
@@ -156,7 +156,7 @@ head(query_table)  %>%
 
 ```{r}
-reticulate::use_python("/usr/local/bin/python3")
+reticulate::use_python("/Users/dao/anaconda3/bin/python")
````
need to revert when done

carlosparadis commented on vignettes/syntax_extractor.Rmd:
```diff
@@ -0,0 +1,513 @@
+conf <- yaml::read_yaml("../conf/maven_py.yml")
+
+# Paths from config
+srcml_path <- tool[["srcml"]]
```
missing get

carlosparadis reviewed (changes requested):
here a couple more, not exhaustive.

daomcgill committed:
i #313 Use getters for config

- Added getter for src_folder
- Updated notebook to use getters

Signed-off-by: Dao McGill <dmcgill@haawaii.edu>

daomcgill committed:
i #313 workflow revert

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #313 Update description

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
Merge branch 'master' of https://github.com/sailuh/kaiaulu

daomcgill committed:
Merge branch 'master' into 313-expand-syntax-extractor

daomcgill committed:
Merge branch '313-expand-syntax-extractor' into 314-externalize-parsed-source-code

daomcgill committed:
i #314 Use get() for execs

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #313 Display tables for notebook

- remove print statement
- gt displays head(10)

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
Merge branch '313-expand-syntax-extractor' into 314-externalize-parsed-source-code

daomcgill committed:
i #314 Config changes

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #313 Config changes

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #313 Minor fixes for notebook

- Added back filters using get()

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
Merge branch '313-expand-syntax-extractor' into 314-externalize-parsed-source-code

daomcgill committed:
i #314 Fixed exec script configs

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>

daomcgill committed:
i #314 New config file structure

Signed-off-by: Dao McGill <dmcgill@hawaii.edu>
